builtin/plugin/backend.go

```diff
@@ -49,10 +49,21 @@ func Backend(ctx context.Context, conf *logical.BackendConfig) (logical.Backend,
 
 	sys := conf.System
 
-	// NewBackend with isMetadataMode set to true
-	raw, err := bplugin.NewBackend(ctx, name, pluginType, sys, conf, true)
+	// NewBackend with isMetadataMode set to false
+	raw, err := bplugin.NewBackend(ctx, name, pluginType, sys, conf, false, true)
```

Comment:
I think that this double call can be avoided if we always attempted running v5 within `NewPluginClient` (with the split PluginSet approach).

Reply:
Factory() calls Setup() [here](https://github.com/hashicorp/vault/blob/f0f709478cbd0a92fe17bf66839c6663693c478b/builtin/plugin/backend.go#L33) so I don't think it is necessary.

Reply:
same as the other comment. I had some issues with unseal trying this out.